game-core/src/test/java/games/strategy/engine/data/RouteFinderTest.java

```diff
@@ -97,7 +102,78 @@ void testNoRouteOnInvalidGraph() {
         .thenReturn(Collections.singleton(territories.get(1)));
     final RouteFinder routeFinder = new RouteFinder(map, t -> true, new ArrayList<>(), player);
     final Optional<Route> optRoute =
-        routeFinder.findRoute(territories.get(0), territories.get(territories.size() - 1));
+        routeFinder.findRouteByDistance(
+            territories.get(0), territories.get(territories.size() - 1));
+    assertFalse(optRoute.isPresent());
+  }
+
+  @Test
+  void testFindRouteByCost() {
+    final RouteFinder routeFinder = new RouteFinder(map, t -> true, new ArrayList<>(), player);
+    final Optional<Route> optRoute =
```

Comment:
Bit of nit, OptionalMatchers test-utility could be used to clean this up a good bit, eg:

```
final RouteFinder routeFinder = new RouteFinder(map, t -> true, new ArrayList<>(), player);

 final Optional<Route> result =
        routeFinder.findRouteByCost(territories.get(0), territories.get(territories.size() - 1));

final Route expectedResult = new Route(Stream.of(0, 3, 6, 8).map(territories::get).collect(Collectors.toList()));
assertThat(result, isPresentAndIs(expectedResult));
```

The 'assertThat' is also a good place to add a commentary string of why we have the expected results, why are territories 0, 3, 6, and 8 the expected territories? It could be reverse engineered, but that level of effort does not scale well (consider a scenario where this is just one of 50 failing tests, knowing why a test is supposed to pass is a best-practice. 

Of some interest, there were even a few master's papers written about testing that looked specifically at TripleA (we did not score very well):

of note, this paragraph: http://www.cs.loyola.edu/~binkley/papers/icsm-12-smells.pdf
> (Page 1) The  second  study  is  a controlled  experiment  involving  twenty  master  students.  Itis  aimed  at  analyzing  whether  the  presence  of  test  smellsaffects  the  comprehension  of  source  code  during  software maintenance.  In  the  study  we  asked  subjects  to  perform different program comprehension tasks and we measured thesubjects’  performance  using  both  correctness  and  the  timespent to perform a task.

> Collected  data  from  the  first  study  shows  that  there  is  a high  diffusion  of  the  test  smells  in  both  open  source  and industrial  software  systems.  In  addition,  the  second  studyprovides  evidence  that  test  smells  have  a  strong  negativeimpact on program comprehension and maintenance.

The other interesting paper that included TripleA if you have not come across it yet: https://testroots.org/assets/papers/athanasiouTSE.pdf

Reply:
Agree that they could be better. I'm gonna skip updating this one since there are already a bunch of existing unit tests in this class and it follows that same pattern. A separate PR to improve the unit tests in this class would be good.

Reply:
Okay, but let's be careful of a "The bureaucracy is expanding to meet the needs of the expanding bureaucracy"